plugins/outputs/splunkhec/splunkhec.go

```diff
@@ -0,0 +1,212 @@
+package splunkhec
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type SplunkHEC struct {
+	Token  string
+	Url string
+    Index string
+    Source string
+
+	Timeout internal.Duration
+	client *http.Client
+}
+
+var sampleConfig = `
+  ## Splunk HEC Token (also used for Data Channel ID)
+  token = "my-secret-key" # required.
+
+  ## Splunk HEC endpoint
+  url = "https://localhost:8088/services/collector" # required.
+
+  ## Splunk Index: Must be a metrics index, must be allowed by above token
+  # index = "telegraf"
+
+  ## Source: Set the 'source' on the events (defaults to: telegraf)
+  # source = "telegraf"
+
+  ## Connection timeout.
+  # timeout = "5s"
+`
+
+
+/* Splunk HEC Structs */
+type HECTimeSeries struct {
+    Time    float64     `json:"time"`
+    Event   string      `json:"event"`
+    Index   string      `json:"index,omitempty"`
+    Source  string      `json:"source,omitempty"`
+    Host    string      `json:"host"`
+    Fields  interface{} `json:"fields"`
+
+}
+
+
+func NewSplunkHEC() *SplunkHEC {
+	return &SplunkHEC{
+	}
+}
+
+func (d *SplunkHEC) Connect() error {
+	if d.Token == "" {
+		return fmt.Errorf("token is a required field for Splunk HEC output")
+	}
+	if d.Url == "" {
+		return fmt.Errorf("url is a required field for Splunk HEC output")
+	}
+
+	d.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: d.Timeout.Duration,
+	}
+	return nil
+}
+
+func (d *SplunkHEC) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+    var hecPostData string
+
+	for _, m := range metrics {
+		if hecMs, err := buildMetrics(m,d); err == nil {
+            hecPostData = hecPostData+string(hecMs)
+        } else {
+			log.Printf("I! unable to build Metric for %s, skipping\n", m.Name())
+		}
+	}
+
+	redactedApiKey := "****************"
```

Comment:
this could be in a constant, so you don t re-create a string at each call of the function.